### PR TITLE
Support is-supported-script expression

### PR DESCRIPTION
**Pull request by ianthetechie**
Closes https://github.com/greensopinion/flutter-vector-map-tiles/issues/113.

I have tested this on a variety of complex shaping scripts such as Nepalese, Burmese, and Khmer, as well as RTL languages such as Arabic and Hebrew. These are the standard problem cases for text rendering, and it appears to pass as a result of not using SDF glyph rendering like most other "modern" renderers, which is pretty cool! As such, the implementation turned out to be pretty simple.

### CODE REVIEW

File: lib/src/themes/expression/is_supported_script_expression.dart
Comment (greensopinion) [inline lib/src/themes/expression/is_supported_script_expression.dart:15]: What does this mean? What is "complex support"?

File: test/src/themes/expression/expression_parser_test.dart
Comment (greensopinion) [inline test/src/themes/expression/expression_parser_test.dart:397]: what is a complex script string?

File: test/src/themes/expression/expression_parser_test.dart
Comment (greensopinion) [inline test/src/themes/expression/expression_parser_test.dart:393]: Great to have some tests! It would be good to have some `false` test cases too.

**Review by greensopinion — COMMENTED** [review]
Thanks for the contribution, it looks great!

I have some questions and a suggestion inline below.

File: test/src/themes/expression/expression_parser_test.dart
Comment (ianthetechie) [reply test/src/themes/expression/expression_parser_test.dart:393]: The comment in `is_supported_script_expression.dart` addresses this, but apologies if that was not clear. At the moment, there are no cases for which this expression would be false, thanks to how rendering is implemented in the vector renderer / flutter map :)

File: lib/src/themes/expression/is_supported_script_expression.dart
Comment (ianthetechie) [reply lib/src/themes/expression/is_supported_script_expression.dart]: I'll update the comment with a Wikipedia link. It's a bit esoteric, but as a native speaker of a language using a Latin-derived script, we naturally assume that, for example, the glyph "a" is always written as "a" but that isn't always true. Complex scripts actually require different rendering of glyphs depending on surrounding context. This goes by a number of names like complex scripts, complex shaping support, complex text layout, etc.: https://en.wikipedia.org/wiki/Complex_text_layout.

The reason that Mapbox / MapLibre have the `is-supported-script` expression is because the "canonical" rendering method for text in that world is such that it can't support complex scripts. Rather each glyph is encoded into a signed distance field which is static by nature and designed to map well into GPU textures for super quick rendering with efficient scale, halo effects, etc. This works super well and has a bunch of cool properties, and even works with Chinese, Japanese, and Korean, but the model totally breaks on a bunch of other (mostly Asian and Indic) scripts where "a" isn't always rendered the same.

(A secondary reason that this `is-supported-script` expression exists is that the JavaScript implementation doesn't support RTL languages like Hebrew and Arabic out-of-the-box, and requires a separate plugin to be loaded... native platforms don't have this issue, so the implementation also needs to account for that.)

I've just pushed an updated comment, but let me know if this requires further explanation or if docs should go in a different place.

Comment (greensopinion) [reply lib/src/themes/expression/is_supported_script_expression.dart]: Thanks for the detailed explanation, that helps a lot.

**Comment by greensopinion** [comment]
The contribution looks great. One of the checks failed, please resolve the formatting issues: https://github.com/greensopinion/dart-vector-tile-renderer/actions/runs/3993332817/jobs/6870719716

**Comment by ianthetechie** [comment]
Sure thing. Just ran `flutter format .` and pushed the changes @greensopinion.

**Comment by greensopinion** [comment]
There are some failing checks [see details](https://github.com/greensopinion/dart-vector-tile-renderer/actions/runs/4003704639/jobs/6901505515)

those will need addressing before we can merge this contribution.
You can run those checks locally on your machine by running the commands listed here: https://github.com/greensopinion/dart-vector-tile-renderer/blob/main/.github/workflows/CI.yaml

**Comment by ianthetechie** [comment]
Thanks for the link. GitHub CI is a bit opaque since it doesn't run checks without manual approval apparently :joy: Sorry for all the back and forth as a result.

**Comment by ianthetechie** [comment]
Just a quick bump on this ;)

**Review by greensopinion — APPROVED** [review]
Thanks for the improvement, and sorry for the delay.

**Pull request merged**
